doc/concepts/configuration.rst

```diff
@@ -0,0 +1,617 @@
+
+For example, ``TT_IPROTO_LISTEN`` and ``TT_IPROTO_LISTEN_DEFAULT`` correspond to the ``iproto.listen`` option.
+``TT_SNAPSHOT_DIR`` and ``TT_SNAPSHOT_DIR_DEFAULT`` correspond to the ``snapshot.dir`` option.
+To see all the supported environment variables, execute the ``tarantool`` command with the ``--help-env-list`` :ref:`option <configuration_command_options>`.
+
+.. code-block:: console
+
+    $ tarantool --help-env-list
+
+Below are a few examples that show how to set environment variables of different types:
+
+*   In the example below, ``TT_IPROTO_LISTEN`` is used to specify a :ref:`listening host and port <configuration_options_connection>` values:
+
+    ..  code-block:: console
+
+        $ export TT_IPROTO_LISTEN='127.0.0.1:3311'
+
+*   To specify several listening addresses, separate them by a comma without space:
+
+    ..  code-block:: console
+
+        $ export TT_IPROTO_LISTEN='127.0.0.1:3311,127.0.0.1:3312'
+
+*   To assign map values to environment variables, use JSON objects.
+    In the example below, ``TT_APP_CFG`` is used to specify the value of a custom configuration property for a :ref:`loaded application <configuration_application>:
+
+    .. code-block:: console
+
+        $ export TT_APP_CFG='{"greeting":"Hi"}'
+
+
+.. NOTE::
+
+    The ``TT_INSTANCE_NAME`` and ``TT_CONFIG`` environment variables can be used to :ref:`run <configuration_run_instance>` the specified Tarantool instance with configuration from the given file.
+
+
+
+
+
+
+.. _configuration_centralized:
+
+Centralized configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+..  admonition:: Enterprise Edition
+    :class: fact
+
+    Centralized configuration is supported by the `Enterprise Edition <https://www.tarantool.io/compare/>`_ only.
+
+Tarantool enables you to store configuration data in one reliable place using etcd.
+To achieve this, you need to:
+
+1.   Put a YAML file with a cluster's configuration to an etcd server:
+
+    .. code-block:: console
+
+        $ etcdctl put /example/config/all.yaml < remote_config.yaml
+
+2.   Provide a local YAML configuration with an etcd endpoint address and key prefix in the ``config`` section:
+
+    ..  literalinclude:: /code_snippets/test/config/etcd.yaml
+        :language: yaml
+        :dedent:
+
+Learn more from the following guide: :ref:`Storing configuration in etcd <configuration_etcd>`.
+
+
+..  _configuration_precedence:
+
+Configuration precedence
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Tarantool configuration options are applied in the following order:
+
+-   `TT_*_DEFAULT` :ref:`environment variables <configuration_environment_variable>`.
+-   :ref:`Centralized configuration <configuration_centralized>` stored in etcd.
+-   Configuration from a :ref:`local YAML file <configuration_file>`.
+-   `TT_*` :ref:`environment variables <configuration_environment_variable>`.
+
+This means that `TT_*` environment variables have a higher priority than other configuration ways.
```

Comment:
How about the reverse order: list the configuration sources from highest priority to lowest one? I guess that it may be more intuitive.

And maybe you'll like the [menthal rule](https://github.com/tarantool/tarantool/blob/10372d0d9ded18ff64c256e125ce740bb0605587/src/box/lua/config/init.lua#L143-L147).

Reply:
Agree, will fix